cli/code_scanner.py

```diff
@@ -531,3 +553,45 @@ def _does_severity_match_severity_threshold(severity: str, severity_threshold: s
         return True
 
     return detection_severity_value >= Severity.try_get_value(severity_threshold)
+
+
+def _get_scan_result(cycode_client, scan_async_result: ScanInitializationResponse,
+                     scan_details: ScanDetailsResponse) -> ZippedFileScanResult:
+    scan_result = ZippedFileScanResult(did_detect=False, detections_per_file=[],
+                                       scan_id=scan_async_result.scan_id,
+                                       report_url=scan_details.report_url)
+    if not scan_details.detections_count:
+        return scan_result
+
+    wait_for_detections_creation(cycode_client, scan_async_result.scan_id, scan_details.detections_count)
+    scan_detections = cycode_client.get_scan_detections_with_paging(scan_async_result.scan_id)
+    scan_result.detections_per_file = _map_detections_per_file(scan_detections)
+    scan_result.did_detect = True
+    return scan_result
+
+
+def wait_for_detections_creation(cycode_client, scan_id: str, expected_detections_count: int):
+    end_polling_time = time.time() + DETECTIONS_COUNT_VERIFICATION_TIMEOUT_IN_SECONDS
+    while time.time() < end_polling_time:
+        scan_persisted_detections_count = cycode_client.get_scan_detections_count(scan_id)
+        if scan_persisted_detections_count == expected_detections_count:
+            return
+        time.sleep(3)
```

Comment:
:exclamation:**Cycode**: SAST violation: 'Arbitrary time.sleep()'.
Severity: High

Would you like to exclude this SAST violation from your status checks?
Tell us what to do with one of the following hashtags:

| Tag | Short Description |
| --- | --- |
| **#cycode_sast_false_positive** | Applies to this SAST violation for all repos in your organization |
| **#cycode_sast_ignore_everywhere** | Applies to this SAST violation for all repos in your organization |
| **#cycode_sast_ignore_here** | Applies to this request only |

 <!-- c07a7373-0af1-49ed-85ea-85586028c007 -->